tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_IPI.py

```diff
@@ -0,0 +1,98 @@
+import logging
+import pytest
+import random
+from ocs_ci.framework.testlib import (
+    tier4, ManageTest, aws_platform_required,
+    ipi_deployment_required, ignore_leftovers)
+from ocs_ci.ocs import machine
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from tests.helpers import wait_for_resource_state
+from tests.sanity_helpers import Sanity
+from ocs_ci.ocs import platform_nodes
+from ocs_ci.ocs.node import add_new_node_and_label_it
+
+
+log = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@tier4
+@aws_platform_required
+@ipi_deployment_required
+class TestAutomatedRecoveryFromFailedNodes(ManageTest):
+    """
+    Knip-678 Automated recovery from failed nodes
+    """
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.mark.parametrize(
+        argnames="failure",
+        argvalues=[
+            pytest.param(
+                *["shutdown"]
+            ),
+            pytest.param(
+                *["terminate"]
+            )
+        ]
+    )
+    def test_automated_recovery_from_failed_nodes_IPI_reactive(
+            self, pvc_factory, pod_factory, failure
+    ):
+        """
+        Knip-678 Automated recovery from failed nodes
+        Reactive case - IPI
+        """
+        # Get the osd associated node
+        osd_pods_obj = pod.get_osd_pods()
+        osd_node_obj = pod.get_pod_node(random.choice(osd_pods_obj))
+        log.info(f"Selected OSD is {osd_node_obj.name}")
+
+        # Get the machine name using the node name
+        machine_name = machine.get_machine_from_node_name(osd_node_obj.name)
+        log.info(f"{osd_node_obj.name} associated machine is {machine_name}")
+
+        # Get the machineset name using machine name
+        machineset_name = machine.get_machineset_from_machine_name(
+            machine_name
+        )
+        log.info(
+            f"{osd_node_obj.name} associated machineset is {machineset_name}"
+        )
+
+        # Add a new node and label it
+        add_new_node_and_label_it(machineset_name)
+
+        # Induce failure
+        aws_obj = platform_nodes.AWSNodes()
```

Comment:
should be used using the 'nodes' fixture from tests/conftest.py

see reference here https://github.com/red-hat-storage/ocs-ci/blob/1839221c2847b5f0d8dbe287e6f0a5f0542c8bec/tests/manage/z_cluster/nodes/test_nodes_restart.py#L71

Reply:
Done